cilium/cmd/policy_trace.go

```diff
@@ -170,33 +146,28 @@ func init() {
 	policyTraceCmd.Flags().StringVarP(&dstK8sPod, "dst-k8s-pod", "", "", "Destination k8s pod ([namespace:]podname)")
 }
 
-func appendEpLabelsToSlice(epID string, labelSlice []string) []string {
-	ep, err := client.EndpointGet(epID)
+func appendIdentityLabelsToSlice(secID string, labelSlice []string) []string {
+	resp, err := client.IdentityGet(secID)
 	if err != nil {
-		Fatalf("Cannot get endpoint corresponding to identifier %s: %s\n", epID, err)
+		Fatalf("%s", err)
 	}
-	labelSlice = append(labelSlice, ep.Identity.Labels...)
+	labelSlice = append(labelSlice, resp.Labels...)
 	return labelSlice
 }
 
-// Returns the labels for security identity ID and an error if the labels cannot be retrieved.
-func getLabelsFromIdentity(ID int64) ([]string, error) {
-	convertedID := policy.NumericIdentity(ID)
-	resp, err := client.IdentityGet(convertedID.StringID())
+func appendEpLabelsToSlice(epID string, labelSlice []string) []string {
+	ep, err := client.EndpointGet(epID)
 	if err != nil {
-		return nil, fmt.Errorf("unable to retrieve labels for security identity %s: %s", convertedID.StringID(), err)
-	}
-	if resp == nil {
-		return nil, fmt.Errorf("security identity %d not found", ID)
+		Fatalf("Cannot get endpoint corresponding to identifier %s: %s\n", epID, err)
 	}
-	return resp.Labels, nil
+	labelSlice = append(labelSlice, ep.Identity.Labels...)
```

Comment:
Just a warning, this line is directly modifying `labelSlice`. Is that intentional?

```
newSlice:= appendEpLabelsToSlice(foo, mySlice)
// mySlice == newSlice is true
```

Reply:
I have the functions now directly return the result of `append`. This leaves it up to the caller whether they want to utilize a new slice or not. In this case, reusing it is what I want to do.